documentation/source/optimizers.md

```diff
@@ -0,0 +1,142 @@
+[RMS_PROP](https://pytorch.org/docs/stable/generated/torch.optim.RMSprop.html#torch.optim.RMSprop)), but also 
+[RMSpropTF](http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf) and 
+[Lamb](https://github.com/NVIDIA/DeepLearningExamples/blob/master/PyTorch/LanguageModeling/Transformer-XL/pytorch/lamb.py).
+
+## Set the optimizer in the code
+Optimizers should be part of the training parameters.
+
+
+```py
+from super_gradients import Trainer
+
+trainer = Trainer(...)
+
+trainer.train(
+    training_params={"optimizer": "Adam", "optimizer_params": {"eps": 1e-3}, ...}, 
+    ...
+)
+```
+
+**Note**:
+The `optimizer_params` is a dictionary of all the optimizer parameters you want to set. It can be any argument defined in the optimizer `__init__` method , except for `params` because this argument corresponds to the model to optimize and is automatically provided by the Trainer.
+
+
+## Set the optimizer in the recipes
+When working with recipes, you need to modify the [recipes/training_hyperparams](https://github.com/Deci-AI/super-gradients/tree/master/src/super_gradients/recipes/training_hyperparams) as below:
+
+```yaml
+# recipes/training_hyperparams/my_training_recipe.yaml
+
+...
+
+optimizer: Adam
+optimizer_params:
+  eps: 1e-3
+```
+
+
+## Use Custom Optimizers
+If your own optimizer is not natively supported by SuperGradients, you can always register it!
+
+```py
+from super_gradients.common.registry.registry import register_optimizer
+
+@register_optimizer()
+class CustomOptimizer:
+    def __init__(
+            self,
+            params, # This arg is the only required regardless of your optimizer, the rest depends on your optimizer. 
+            alpha: float, 
+            betas: float
+    ):
+        defaults = dict(alpha=alpha, betas=betas)
+        super(CustomOptimizer, self).__init__(params, defaults)
+
+    ...
+```
+
+And then update your training hyperparameters:
+
+```yaml
+# my_training_hyperparams.yaml
+
+...
+
+optimizer: CustomOptimizer
+optimizer_params:
+  alpha: 1e-3
+  betas: 1e-3
+```
+
+## Customize learning rate for different model blocks 
+You can define the learning rate to use on each section of your model by working with `initialize_param_groups` and `update_param_groups`.
+- `initialize_param_groups` defines the groups, and the learning rate to use for each group. It is called on instantiation.
+- `update_param_groups` updates the learning rate for each group. It is called by LR callbacks (such as `LRCallbackBase`) during the training. 
+
+If your model (i.e. any `torch.nn.Module`) is lacking these methods, the same learning rate will be applied to every block.
+But if you implement them, it will be taken into account by the Trainer just like with any other SuperGradients model.
+
+#### Example
+Let's imagine that you have your own custom model and that you want work with a different learning rate on the backbone.
```

Comment:
Instead of "let's imagine that you ... ". please write "Assume you have your own..."

Reply:
Changed in d66358e09ade583670912a7c5815a0d7cbd4300a